File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.py
class Solution:
    def gcd(self, o: int,e: int) -> int:
        if e==0:
            return o
        return self.gcd(e, o%e)
    def gcdOfOddEvenSums(self, n: int) -> int:
        oddsum = 0
        evensum = 0
        i = 1
        while (i<=2*n):
            if i%2==0:
             evensum+=i
            else:
             oddsum+=i
            i+=1
        ans = gcd(oddsum,evensum)
        return ans

        

File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.java
class Solution {
    public int gcd(int a,int b){
        if(b==0) return a;
        return gcd(b,a%b);
    }
    public int gcdOfOddEvenSums(int n) {
        int osum=0;
        int esum=0;
        int i=1;
        while(i<=2*n){
            if(i%2==0){
                esum+=i;
            }else{
                osum+=i;
            }
            i++;
        }
      
        return gcd(esum,osum);
    }
}